phygnn/layers/custom_layers.py

```diff
@@ -1527,23 +1539,42 @@ def call(self, x, hi_res_feature=None):
         hi_res_feature : tf.Tensor | np.ndarray
             This should be a 4D array for spatial enhancement model or 5D array
             for a spatiotemporal enhancement model (obs, spatial_1, spatial_2,
-            (temporal), 1). This is NaN where there are no observations and
-            real values where observations exist.
+            (temporal), features). This is NaN where there are no observations
+            and real values where observations exist.
+        exo_data : tf.Tensor | np.ndarray
+            This is an array of exogenous data used to imform the embedding,
+            like topography
 
         Returns
         -------
         x : tf.Tensor
-            Output tensor with embedded hi_res_feature and observation mask
-            concatenated to input.
+            Output tensor with embedding concatenated to input.
         """
         if hi_res_feature is None:
-            hi_res_feature = tf.fill(x[..., :1].shape, np.nan)
+            hr_shape = (*x[..., 0].shape, len(self.features))
+            hi_res_feature = tf.constant(np.nan, shape=hr_shape, dtype=x.dtype)
```

Comment:
I'm confused as to why this would ever be called with `hi_res_feature=None`... Isn't the whole point of this method to integrate this input?

Also confused why you'd allow `exo_data=None if len(self.exo_features) > 0`? What would this use case be?

Reply:
Yeah good question. It's so I can use the model with and without these extra inputs. Train with obs and then do inference with different sets of inputs.

Reply:
Gotcha that makes sense, i think definitely include a note on this in the docstrings esp since None is the default for these args